executor/executor.go

```diff
@@ -328,9 +330,43 @@ func (executor *Executor) execute(
 			}
 			return response, nil
 		}
-		msg := "decision executed successfully"
 
-		executor.logger.Infof(ctx, msg)
+		// short pooling to trigger pod status with max 15 minutes
+
+		statusMap := make(map[string]string)
+		statusMap["Running"] = "pod restarted successfully"
+		statusMap["Failed"] = "pod failed to restart"
+		statusMap["Unknown"] = "pod status is unknown"
+		statusMap["PodInitializing"] = "pod restarting"
+
+		backoff := 15 * time.Minute
+		msg := "pod restarting exceeded timout (15 min)"
+		start := time.Now()
+		timeout := int(backoff.Seconds())
+
+		for time.Now().Second() - start.Second() < timeout {
+
+			status := "Pending"
+
+			time.Sleep(15 * time.Second)
+			pods, _ := executor.kube.GetNameSpacePods(namespace)
+
+			for _, pod := range pods.Items {
```

Comment:
if a controller/service has multiple replicas the below logic is kinda will be wrong with the first check because based on the `RollingUpdateStrategy` it will determine how many from the replicas will be updated before the all of replicas been updating and if that the case how you gaurntee that the first replica with the same name you got is not from old replicaset? do you query from the api server with sort arg or something?
@mohamedveron

Reply:
when you make change in deployment it will generate new replica and replace the current pods with new pods and all these pods will end up with the same status so if one of these pods not restarted successfully it will be the same case with the other pods

Reply:
the only corner case which will break when one of the pods in the replicaset can't be allocated in  some node

Reply:
beside the corner state, do you query the pods from the latest RS?
To know more what I mean by RollingUpdateStrategy and how it would impact the current flow please read the below article 
https://www.weave.works/blog/kubernetes-deployment-strategies

@mohamedveron